### PR TITLE
M500 ingestion

### PR DESCRIPTION
**Pull request by WWF-ConsEvidence**

### CODE REVIEW

File: src/api/ingest/utils.py [hunk 1]
```diff
@@ -75,6 +76,9 @@ def import_table(datafile, identifier, cleardata, tofile):
             print("There was an error in importing row {}. Import failed. See details below.".format(rowcount))
             print(e.args[0])
             sys.exit()
+        except ValueError as e:
```
Comment (saanobhaai) [inline src/api/ingest/utils.py:88]: You should be able to do, on line 75:
`except IntegrityError, ValueError as e:`

**Review by saanobhaai — COMMENTED** [review]
one small comment

File: src/api/ingest/utils.py (hunk 1 above)
Comment (WWF-ConsEvidence) [reply src/api/ingest/utils.py:88]: Ok!

File: src/api/signals.py [hunk 2]
```diff
@@ -14,14 +14,14 @@ def create_or_update_user_profile(sender, instance, created, **kwargs):
...
-    year = instance.interviewyear
-    upsert_mpa_interviewyear(mpa, year)
-
-
-post_save.connect(
-    update_mpa_interviewyears,
-    sender=Household,
-    dispatch_uid=f"{Household._meta.object_name}_update_mpa_interviewyears",
-)
+# def update_mpa_interviewyears(sender, instance, created, **kwargs):
```
Comment (saanobhaai) [inline src/api/signals.py]: We'll need this, so should be uncommented before merging...

File: src/api/ingest/utils.py [hunk 3]
```diff
@@ -699,149 +36,58 @@ def import_household(datafile, identifier, cleardata, tofile):
...
+                try:
+                    datetime.strptime(cleaned_row[fld_date.name], "%m/%d/%Y")
+                    cleaned_row[fld_date.name] = datetime.strptime(cleaned_row[fld_date.name], "%m/%d/%Y").strftime("%Y-%m-%d")
+                except:
+                    #its not in MM-DD-YYYY
+                    pass
+        # TODO - how to identify boolean fields and make them python booleans
+        for fld_truth in themodel._meta.get_fields():
+            if fld_truth.__class__.__name__ == "BooleanField":
+                cleaned_row[fld_truth.name] = mpatruth[cleaned_row[fld_truth.name]]
```
Comment (saanobhaai) [inline src/api/ingest/utils.py:80]: This seems reasonable... but might be good to also stipulate to collectors somewhere that values should be 0/1 (or whatever)

File: src/api/ingest/utils.py [hunk 4]
```diff
@@ -699,149 +36,58 @@ def import_household(datafile, identifier, cleardata, tofile):
...
-        if newrow["lonfrac"] == None:
-            newrow["lonfrac"] = 0
-        if newrow["latfrac"] == None:
-            newrow["latfrac"] = 0
         try:
-            Household.objects.create(**newrow)
-        except (IntegrityError):
-            print("Duplicate key: {}. Import failed.".format(row["householdid"]))
+            themodel.objects.create(**cleaned_row)
+        except IntegrityError as e:
```
Comment (saanobhaai) [inline src/api/ingest/utils.py:84]: Are there other exceptions to catch here?

File: src/api/ingest/utils.py [hunk 5]
```diff
@@ -699,149 +36,58 @@ def import_household(datafile, identifier, cleardata, tofile):
...
-            row[thefld[0]] = (
-                thechoices["data__" + thefld[1]][row[thefld[0]]]
-                if row[thefld[0]]
-                else None
-            )
+        #convert date fields to YYYY-MM-DD if they are in MM-DD-YYYY
+        for fld_date in themodel._meta.get_fields():
+            if fld_date.__class__.__name__ == "DateField":
+                try:
+                    datetime.strptime(cleaned_row[fld_date.name], "%m/%d/%Y")
```
Comment (saanobhaai) [inline src/api/ingest/utils.py:72]: At one point we were talking about separating the csv columns into month,date,year -- that would be preferable to relying on a particular string pattern.

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Saw a few more things worth looking at... in addition, we should ignore and stop tracking all `.DS_Store` files.

File: src/api/ingest/utils.py (hunk 1 above)
Comment (WWF-ConsEvidence) [reply src/api/ingest/utils.py:88]: Hold on, I wanted to give different messages for these but maybe I can just leave it with the generic message

File: src/api/ingest/utils.py (hunk 3 above)
Comment (WWF-ConsEvidence) [reply src/api/ingest/utils.py:80]: Ok

**Pull request closed without merging**